server/routes.go

```diff
@@ -684,13 +763,15 @@ func Serve(ln net.Listener, allowOrigins []string) error {
 	r.POST("/api/copy", CopyModelHandler)
 	r.DELETE("/api/delete", DeleteModelHandler)
 	r.POST("/api/show", ShowModelHandler)
+	r.POST("/api/blobs/:digest", CreateBlobHandler)
 
 	for _, method := range []string{http.MethodGet, http.MethodHead} {
 		r.Handle(method, "/", func(c *gin.Context) {
 			c.String(http.StatusOK, "Ollama is running")
 		})
 
 		r.Handle(method, "/api/tags", ListModelsHandler)
+		r.Handle(method, "/api/blobs/:digest/path", GetBlobHandler)
```

Comment:
`HEAD /api/blobs/:diget` ?

Reply:
`HEAD /api/blobs/:digest` would definitely be preferable here. It wouldn't leak the path to the blob either, just test its existence.

Reply:
The purpose of this handler is to return the path to the blob so HEAD wouldn't cut it